pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +261,106 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+// getAPIServicesToManage gets the desired list of API Services that will be managed by this operator
+// note that some services might be managed by an external operators/servers
+func (c *APIServiceController) getAPIServicesToManage() ([]*apiregistrationv1.APIService, error) {
+	if externalOperatorPrecondition := c.externalOperatorPreconditionFunc(); externalOperatorPrecondition != nil {
+		return nil, externalOperatorPrecondition
+	}
+
+	filteredApiServices := []*apiregistrationv1.APIService{}
+
+	// the c.initialAPIServices list is authoritative
+	// it can compete with oauth-server in case of errors but it could save us on downgrade (the current coao and an old cao)
+	for _, apiService := range c.initialAPIServices {
+		if c.isAPIServicePotentiallyManagedByExternalServer(apiService) && c.isAPIServiceAnnotatedByExternalServer(apiService) {
+			continue
+		}
+		filteredApiServices = append(filteredApiServices, apiService)
+	}
+
+	{
+		added, removed := diffAPIServices(c.managedAPIServices, filteredApiServices)
+		if len(added) > 0 {
+			c.eventRecorder.Eventf("OAuthAPIResourcesManagement", "The API Services for the following GVs %v will be managed by this operator (CAS-O)", added)
+			klog.V(2).Infof("The API Services for the following GVs %v will be managed by this operator (CAS-O)", added)
+		}
+
+		if len(removed) > 0 {
+			c.eventRecorder.Eventf("OAuthAPIResourcesManagement", "The API Services for the following GVs %v will be managed by an external operator/server", removed)
+			klog.V(2).Infof("The API Services for the following GVs %v will be managed by an external operator/server", removed)
+		}
+	}
+
+	c.managedAPIServices = filteredApiServices
+	return c.managedAPIServices, nil
+}
+
+func (c *APIServiceController) isAPIServicePotentiallyManagedByExternalServer(apiService *apiregistrationv1.APIService) bool {
+	for _, apiGroupManagedByOAuthServer := range c.apiGroupsManagedByExternalServer {
+		if apiService.Spec.Group == apiGroupManagedByOAuthServer.Group && apiService.Spec.Version == apiGroupManagedByOAuthServer.Version {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *APIServiceController) isAPIServiceAnnotatedByExternalServer(apiService *apiregistrationv1.APIService) bool {
+	existingApiService, err := c.apiregistrationv1Client.APIServices().Get(apiService.Name, metav1.GetOptions{})
+	if err != nil {
```

Comment:
not found needs different handling.  Assume ownership if not found.

Reply:
we will take over https://github.com/openshift/cluster-openshift-apiserver-operator/pull/294/files#diff-7c7cf45105d9733d46a21001a29c9a1eR42